launch/launch/substitutions/string_join_substitution.py

```diff
@@ -34,13 +36,31 @@ class StringJoinSubstitution(Substitution):
 
     .. code-block:: python
 
-        StringJoinSubstitution(
-            [['https', '://'], LaunchConfiguration('subdomain')], 'ros', 'org'],
+        subdomain = LaunchConfiguration(variable_name='subdomain', default='docs')
+        url = StringJoinSubstitution(
+            [['https', '://'], subdomain], 'ros', 'org'],
```

Comment:
The square brackets are unbalanced. Looking at the XML and YAML examples below, I think this should be:

```suggestion
            [['https', '://', subdomain], 'ros', 'org'],
```